homeassistant/components/streamlabswater/binary_sensor.py

```diff
@@ -1,84 +1,54 @@
 from __future__ import annotations
 
-from datetime import timedelta
-
-from streamlabswater.streamlabswater import StreamlabsClient
-
 from homeassistant.components.binary_sensor import BinarySensorEntity
+from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
-from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
-from homeassistant.util import Throttle
-
-from . import DOMAIN as STREAMLABSWATER_DOMAIN
-
-DEPENDS = ["streamlabswater"]
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
 
-MIN_TIME_BETWEEN_LOCATION_UPDATES = timedelta(seconds=60)
+from . import StreamlabsCoordinator
+from .const import DOMAIN
+from .coordinator import StreamlabsData
 
-ATTR_LOCATION_ID = "location_id"
 NAME_AWAY_MODE = "Water Away Mode"
 
 
-def setup_platform(
+async def async_setup_entry(
     hass: HomeAssistant,
-    config: ConfigType,
-    add_devices: AddEntitiesCallback,
-    discovery_info: DiscoveryInfoType | None = None,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
 ) -> None:
-    """Set up the StreamLabsWater mode sensor."""
-    client = hass.data[STREAMLABSWATER_DOMAIN]["client"]
-    location_id = hass.data[STREAMLABSWATER_DOMAIN]["location_id"]
-    location_name = hass.data[STREAMLABSWATER_DOMAIN]["location_name"]
+    """Set up Streamlabs water binary sensor from a config entry."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
 
-    streamlabs_location_data = StreamlabsLocationData(location_id, client)
-    streamlabs_location_data.update()
+    entities = []
 
-    add_devices([StreamlabsAwayMode(location_name, streamlabs_location_data)])
+    for location_id in coordinator.data.values():
+        entities.append(StreamlabsAwayMode(coordinator, location_id))
 
+    async_add_entities(entities)
 
-class StreamlabsLocationData:
-    """Track and query location data."""
-
-    def __init__(self, location_id: str, client: StreamlabsClient) -> None:
-        """Initialize the location data."""
-        self._location_id = location_id
-        self._client = client
-        self._is_away = None
-
-    @Throttle(MIN_TIME_BETWEEN_LOCATION_UPDATES)
-    def update(self) -> None:
-        """Query and store location data."""
-        location = self._client.get_location(self._location_id)
-        self._is_away = location["homeAway"] == "away"
 
-    def is_away(self) -> bool | None:
-        """Return whether away more is enabled."""
-        return self._is_away
-
-
-class StreamlabsAwayMode(BinarySensorEntity):
+class StreamlabsAwayMode(CoordinatorEntity[StreamlabsCoordinator], BinarySensorEntity):
     """Monitor the away mode state."""
 
-    def __init__(
-        self, location_name: str, streamlabs_location_data: StreamlabsLocationData
-    ) -> None:
+    def __init__(self, coordinator: StreamlabsCoordinator, location_id: str) -> None:
```

Comment:
Should we not make a device and fix the name directly?

Reply:
Wanted to leave that for a follow up, but I don't mind doing it here